File: process.py

# The Process class represents a process in our system.

# When we print processes, we will print them in json format
import json

class Process:

    # Contructor - the default values for the member variables
    # are as indicated - an empty string for the process name,
    # and -1 for the process id
    def __init__(self, process_name="", process_id=-1):
        # Implement the constructor here

    
    # Set the process name 
    def setProcessName(self, process_name):
        # Add implementation here

    # Set the process ID
    def setProcessId(self, process_id):
        # Add implementation here

    # get the process name
    def processName(self):
        # Add implementation here
    
    # get the process id
    def processId(self):
        # Add implementation here

    # This method is called when print() is called. 
    # It should convert the contents of the object to json, and then dump the json format. 
    #
    # # See https://docs.python.org/3/library/json.html for documentation on using JSON
    # in Python.
    # Use the json.dumps() method to print your data, using Compact Encoding.
    #
    # The resulting output should look like this:
    #
    # {"process_id":1,"process_name":"process1"}
    # 
    def __str__(self):
        return json.dumps({"process_name" : self.process_name,
                           "process_id" : self.process_id}, 
                           separators = (",",":"), sort_keys = True)

File: process.cpp
// process.cpp
//
// Include your implementation of the Process class here.

#include <string>
#include <iostream>
#include <sstream>

#include "process.h"
#include "nlohmann/json.hpp"

using namespace std;
 
Process::Process(const string& processName, int processId) 
{
    // Implement the constructor for the Process class here
}


const string& Process::processName() const
{
    // Add implementation here
}

void Process::setProcessName(const string& processName)
{
    // Add implementation here
}

const int Process::processId() const
{
    // Add implementation here
}

void Process::setProcessId(int processId)
{
    // Add implementation here
}


// Friend function to print the object.
// See https://www.learncpp.com/cpp-tutorial/overloading-the-io-operators/ for some 
// instructions on how to implement an operator<< function for your class.
//
// This function should convert the Process object to json (see to_json below), and then 
// print the json object to the output stream. 
//
// The resulting output should look like this:
//
// {"process_id":1,"process_name":"process1"}
//
std::ostream& operator<<(std::ostream& os, const Process& p)
{


    return os;
}

// Populate the given json object with the members of the Process object.
// See https://github.com/nlohmann/json/blob/develop/README.md#arbitrary-types-conversions
void to_json(nlohmann::json& j, const Process& p)
{   
}

// Given an input json object containing the keywords representing a process,
// populate the given Process object.
// See https://github.com/nlohmann/json/blob/develop/README.md#arbitrary-types-conversions

void from_json(const nlohmann::json& j, Process& p)
{
}